Merge drawText-feature into dev

Vargen18 opened the pull request.

Vargen18 committed:
Changed language level to 9, requires java 11. 

Needed for private methods in interfaces.

Vargen18 committed:
Added TextObserver and TextObservable interfaces.

Vargen18 committed:
Implemented observer interfaces.

Vargen18 committed:
Implemented observer interfaces.

Vargen18 committed:
Merge remote-tracking branch 'origin/drawText-feature' into drawText-feature

Vargen18 committed:
Added a class containing the necessary information to print text.

Vargen18 committed:
Argument class -> Text instead of String.

Vargen18 committed:
Reverted to language level 8

Vargen18 committed:
Formatting and unused variable.

Vargen18 committed:
Implemented more TextObservable methods.

Vargen18 committed:
Calls the correct method now

Vargen18 committed:
Add the view as an observer when you create the ingamecontroller. 

Note: You should add view as an observer when you switch level.

Vargen18 committed:
Bugfix, no longer removes observer when you attempt to add, and vice versa

Vargen18 committed:
View now implements TextObserver (which could be questionable)

Vargen18 committed:
Fixed bugs in Level, Dog can now talk!

Vargen18 committed:
Now the dog can actually talk, forgot to implement drawing the text.

Vargen18 committed:
Merged dev into drawText-feature

Jononannes committed:
Added documents for peer review

Vargen18 committed:
Refactored update method to public

Vargen18 committed:
Refactored level to take Player as a parameter

Vargen18 committed:
Fixed incorrect test

Vargen18 committed:
Finished empty method

Vargen18 committed:
You can now tell TextObservers to stop acting upon a former request (i.e. stop drawing text)

Vargen18 committed:
Updated documentation, added a TODO

Vargen18 committed:
Corrected the name of a value in the TextType enum.

Jononannes reviewed (changes requested):
Implement interaction prompts when player is intersecting another GameObject.

Vargen18 committed:
Made Game update the model instead of controller (slims the controller)

Vargen18 committed:
Added abstract method promptInteraction to GameObject. Implemented interaction prompts for some GameObjects.

Vargen18 committed:
Now prompts interaction for each GameObject intersecting with the player

Note: Bugs ahoy! Items don't stop displaying interaction prompt when picked up or they stop intersecting the player.

Vargen18 committed:
Added field to track if interaction has been prompted or not.

Vargen18 committed:
Now doesn't prompt interaction if interaction has already been prompted.

Jononannes committed:
Fixed ConcurrentModificationException

Jononannes committed:
Remove interaction prompt when leaving gameObject or interacting with item

Jononannes committed:
Merge remote-tracking branch 'origin/drawText-feature' into drawText-feature

# Conflicts:
#	src/model/Game.java

Jononannes committed:
Merge remote-tracking branch 'origin/drawText-feature' into drawText-feature

# Conflicts:
#	src/model/Game.java

Jononannes committed:
Merge branch 'drawText-feature' of https://github.com/streckgumman/tda367 into drawText-feature

Vargen18 committed:
Utilized isInteractionPrompted to not prompt interaction when it's already been done. 

This also resolves the NPCs interaction prompts overruling their speech.

Vargen18 committed:
Merge branch 'dev' into drawText-feature

# Conflicts:
#	src/controller/InGameController.java
#	src/model/GameObject.java
#	src/model/levels/Level1.java
#	src/model/levels/Level3.java
#	src/model/npcs/DogWithGun.java
#	src/model/puzzles/Door.java
#	src/view/InGameView.java
